qiskit/algorithms/observables_evaluator.py

```diff
@@ -0,0 +1,130 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Evaluator of auxiliary operators for algorithms."""
+
+from typing import Tuple, Union, List, Sequence, Optional
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.opflow import (
+    PauliSumOp,
+)
+from ..primitives import Estimator, EstimatorResult
+from ..quantum_info.operators.base_operator import BaseOperator
+
+
+def eval_observables(
+    estimator: Estimator,
+    quantum_state: QuantumCircuit,
+    observables: Sequence[Union[BaseOperator, PauliSumOp]],
```

Comment:
I think a lot of value for this function would come from this handling `ListOrDict` used in VQE & co 🙂